pyproject.toml

```diff
@@ -39,17 +39,17 @@ dependencies = [
     "httpx>=0.25.0",
     "importlib-metadata>=4.8.3;python_version<\"3.10\"",
     "importlib-resources>=1.4;python_version<\"3.9\"",
-    "ipykernel",
+    "ipykernel>=6.5.0",
     "jinja2>=3.0.3",
     "jupyter_core",
     "jupyter_server>=2.4.0,<3",
     "jupyter-lsp>=2.0.0",
     "jupyterlab_server>=2.19.0,<3",
     "notebook_shim>=0.2",
     "packaging",
-    "tomli;python_version<\"3.11\"",
+    "tomli>=1.2.2;python_version<\"3.11\"",
     "tornado>=6.2.0",
-    "traitlets",
+    "traitlets>=3.0",
```

Comment:
We almost certainly need a newer version of `traitlets` but pinning it to anything solves the resolution issue, probably because the very first version did not have pins or something, see https://github.com/jupyterlab/maintainer-tools/issues/230. Of note the oldest version with a changelog is 4.0 (https://traitlets.readthedocs.io/en/stable/) so I doubt pinning to 3.0 would inconvenient anyone with Python version modern enough to install JupyterLab.